tests/api/runtimes/client_side/test_auto_mount.py

```diff
@@ -0,0 +1,95 @@
+import os
+import pathlib
+import sys
+
+import pytest
+from fastapi.testclient import TestClient
+from sqlalchemy.orm import Session
+
+from mlrun import code_to_function
+from mlrun.config import config as mlconf
+from mlrun.runtimes.kubejob import KubejobRuntime
+
+from ..base import TestRuntimeBase
+
+
+class TestAutoMountClientSide(TestRuntimeBase):
+    def custom_setup(self):
+        self.image_name = "test/image:latest"
+        self.code_handler = "test_func"
+
+        os.environ["V3IO_ACCESS_KEY"] = self.v3io_access_key = "1111-2222-3333-4444"
+        os.environ["V3IO_USERNAME"] = self.v3io_user = "test-user"
+
+    def _generate_kubejob_runtime(self):
+        runtime = KubejobRuntime()
+        runtime.spec.image = self.image_name
+        return runtime
+
+    def _generate_nuclio_runtime(self):
+        runtime = code_to_function(
+            name=self.name,
+            project=self.project,
+            filename=self.code_filename,
+            handler=self.code_handler,
+            kind="nuclio",
+            image=self.image_name,
+            description="test function",
+        )
+        return runtime
+
+    def _execute_run(self, runtime, **kwargs):
+        runtime.run(
+            name=self.name,
+            project=self.project,
+            artifact_path=self.artifact_path,
+            watch=False,
+        )
+
+    @property
+    def assets_path(self):
+        return (
+            pathlib.Path(sys.modules[self.__module__].__file__).absolute().parent.parent
+            / "assets"
+        )
+
+    @pytest.mark.parametrize("cred_only", [True, False])
+    def test_auto_mount_v3io(
+        self, db: Session, client: TestClient, cred_only, rundb_mock
+    ):
```

Comment:
The reason to put the other runtimes tests under the `tests/api/runtimes` was:
1. to use the api testing facilities - the `db` and `client` fixtures
2. because it practically testing the api side

Now we're testing client side, and it doesn't look like you're using those fixtures so I suggest:
1. Moving the fixtures to `tests/common_fixtures.py`
2. Separating the tests here - kubejob tests under `tests/runtimes/test_base.py`
3. Nuclio tests under `tests/runtimes/test_function.py`